XcodeServerSDK/Server Entities/Repository.swift

```diff
@@ -0,0 +1,148 @@
+//
+//  Repository.swift
+//  XcodeServerSDK
+//
+//  Created by Mateusz Zając on 28.06.2015.
+//  Copyright © 2015 Honza Dvorsky. All rights reserved.
+//
+
+import Foundation
+
+public class Repository: XcodeRead {
+    
+    /**
+    Enumeration describing HTTP access to the repository
+    
+    - None:      No users are not allowed to read or write
+    - LoggedIn:  Logged in users are allowed to read and write
+    */
+    public enum HTTPAccessType: Int {
+        
+        case None = 0
+        case LoggedIn
+        
+        public func toString() -> String {
+            switch self {
+            case .None:
+                return "No users are not allowed to read or write"
+            case .LoggedIn:
+                return "Logged in users are allowed to read and write"
+            }
+        }
+        
+    }
+    
+    /**
+    Enumeration describing HTTPS access to the repository
+    
+    - SelectedReadWrite:         Only selected users can read and/or write
+    - LoggedInReadSelectedWrite: Only selected users can write but all logged in can read
+    - LoggedInReadWrite:         All logged in users can read and write
+    */
+    public enum SSHAccessType: Int {
+        
+        case SelectedReadWrite = 0
+        case LoggedInReadSelectedWrite
+        case LoggedInReadWrite
+        
+        public func toString() -> String {
+            switch self {
+            case .SelectedReadWrite:
+                return "Only selected users can read and/or write"
+            case .LoggedInReadSelectedWrite:
+                return "Only selected users can write but all logged in can read"
+            case .LoggedInReadWrite:
+                return "All logged in users can read and write"
+            }
+        }
+        
+    }
+    
+    public let name: String
+    public var httpAccess: HTTPAccessType = HTTPAccessType.None
+    public var sshAccess: SSHAccessType = SSHAccessType.SelectedReadWrite
```

Comment:
`SSHAccessType.SelectedReadWrite` is default and yes that means that nobody can access this repository if you pass empty arrays - strange but that's what I get in the response from XCS so I even double checked that 😜

Yeah, we can make `LoggedInReadWrite` as a default but I'll leave a comment if anyone will ever look for why we're not strict about this with XCS.